### PR TITLE
Lifecycle update and implementations

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Updated Overwatch rev

Implemented lifecycle for services:

- [x] Consensus
- [ ] ~~http~~ Should be added to new api service instead
- [x] log
- [x] mempool
- [x] network
- [x] storage

### CODE REVIEW

**Comment by danielSanchezQ** [comment]
I'm not really implementing but leaving the place where we should add features for lifecycle management in services. I'm just making them stop on `Shutdown` and `Kill`.
If you feel it is too big, I can leave it here and implement the remaining services in different PRs.

File: nomos-services/consensus/src/lib.rs [hunk 1]
```diff
@@ -389,6 +396,21 @@ where
     DaPoolAdapter: MempoolAdapter<Item = DaPool::Item, Key = DaPool::Key> + Send + Sync + 'static,
     Storage: StorageBackend + Send + Sync + 'static,
 {
+    async fn handle_lifecycle_message(message: LifecycleMessage) -> bool {
```
Comment (zeegomo) [inline nomos-services/consensus/src/lib.rs:399]: it's not clear from the signature what the bool in the return value means

Comment (danielSanchezQ) [reply nomos-services/consensus/src/lib.rs:399]: Yes, I have that itch as well. I will switch it for an enum.

Comment (zeegomo) [reply nomos-services/consensus/src/lib.rs:399]: or maybe rename it something like `should_stop_service`

Comment (danielSanchezQ) [reply nomos-services/consensus/src/lib.rs:399]: > or maybe rename it something like `should_stop_service`

Yep, that would do as well. Thanks!

File: nomos-services/data-availability/src/lib.rs [hunk 2]
```diff
@@ -111,69 +183,28 @@ where
...
-{
-    match msg {
-        DaMsg::RemoveBlobs { blobs } => {
-            futures::stream::iter(blobs)
-                .for_each_concurrent(None, |blob| async move {
-                    if let Err(e) = backend.remove_blob(&blob).await {
-                        tracing::debug!("Could not remove blob {blob:?} due to: {e:?}");
+                Some(msg) = lifecycle_stream.next() => {
+                    if Self::should_stop_service(msg).await {
+                        break;
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/src/lib.rs:201]: This is so confusing in the diff:

```rust
    async fn run(self) -> Result<(), DynError> {
        let Self {
            mut service_state,
            backend,
            da,
            network_relay,
        } = self;

        let network_relay = network_relay
            .connect()
            .await
            .expect("Relay connection with NetworkService should succeed");

        let adapter = Network::new(network_relay).await;
        let mut network_blobs = adapter.blob_stream().await;
        let mut lifecycle_stream = service_state.lifecycle_handle.message_stream();
        loop {
            tokio::select! {
                Some(blob) = network_blobs.next() => {
                    if let Err(e) = Self::handle_new_blob(&da, &backend, &adapter, blob).await {
                        tracing::debug!("Failed to add a new received blob: {e:?}");
                    }
                }
                Some(msg) = service_state.inbound_relay.recv() => {
                    if let Err(e) = Self::handle_da_msg(&backend, msg).await {
                        tracing::debug!("Failed to handle da msg: {e:?}");
                    }
                }
                Some(msg) = lifecycle_stream.next() => {
                    if Self::should_stop_service(msg).await {
                        break;
                    }
                }
            }
        }
        Ok(())
    }
}
```

**Comment by danielSanchezQ** [comment]
Clippy lint is unrelated. Should be fixed in an external PR

**Review by zeegomo — APPROVED** [review]

**Pull request merged**